### PR TITLE
Support separate client side MTLS for production and sandbox endpoints

### PR DESCRIPTION
**Pull request by RusJaI**
Backend implementation for https://github.com/wso2/api-manager/issues/2433

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/authenticator/MutualSSLAuthenticator.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public class MutualSSLAuthenticator implements Authenticator {
     private boolean isMandatory;
 
     // <UniqueIdentifierName,Tier> -Format
-    private HashMap<String, String> certificates;
+    private HashMap<String, HashMap<String, String>> certificates;
```
Comment (RusJaI) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/authenticator/MutualSSLAuthenticator.java:63]: Map<String, Map>

Comment (RusJaI) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/authenticator/MutualSSLAuthenticator.java]: this was fixed

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java [hunk 2]
```diff
@@ -835,7 +836,7 @@ int addClientCertificate(String userName, ApiTypeWrapper apiTypeWrapper, String
      * 4 : If certificate is not found in the trust store.
      * @throws APIManagementException API Management Exception.
      */
-    int deleteClientCertificate(String userName, ApiTypeWrapper apiTypeWrapper, String alias)
+    int deleteClientCertificate(String userName, ApiTypeWrapper apiTypeWrapper, String alias, String keyTye)
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java:839]: Typo in the keyType

**Comment by npamudika** [comment]
Please retrigger the builds as well

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java (hunk 2 above)
Comment (RusJaI) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java]: fixed

**Review by npamudika — DISMISSED** [review]

**Review by npamudika — APPROVED** [review]

**Pull request merged**